src/transformers/pipelines/document_question_answering.py

```diff
@@ -0,0 +1,429 @@
+                if self.tokenizer.cls_token_id is not None:
+                    cls_indices = np.nonzero(np.array(input_ids_span_idx) == self.tokenizer.cls_token_id)[0]
+                    for cls_index in cls_indices:
+                        p_mask[span_idx][cls_index] = 0
+
+            # For each span, place a bounding box [0,0,0,0] for question and CLS tokens, [1000,1000,1000,1000]
+            # for SEP tokens, and the word's bounding box for words in the original document.
+            bbox = []
+            for batch_index in range(num_spans):
+                for i, s, w in zip(
+                    encoding.input_ids[batch_index],
+                    encoding.sequence_ids(batch_index),
+                    encoding.word_ids(batch_index),
+                ):
+                    if s == 1:
+                        bbox.append(boxes[w])
+                    elif i == self.tokenizer.sep_token_id:
+                        bbox.append([1000] * 4)
+                    else:
+                        bbox.append([0] * 4)
+
+            if self.framework == "tf":
+                raise ValueError("Unsupported: Tensorflow preprocessing for DocumentQuestionAnsweringPipeline")
+            elif self.framework == "pt":
+                encoding["bbox"] = torch.tensor([bbox])
+
+            word_ids = [encoding.word_ids(i) for i in range(num_spans)]
+
+        return {
+            **encoding,
+            "p_mask": p_mask,
+            "word_ids": word_ids,
+            "words": words,
+        }
+
+    def _forward(self, model_inputs):
+        p_mask = model_inputs.pop("p_mask", None)
+        word_ids = model_inputs.pop("word_ids", None)
+        words = model_inputs.pop("words", None)
+
+        if isinstance(self.model, VisionEncoderDecoderModel):
+            model_outputs = self.model.generate(**model_inputs)
+        else:
+            model_outputs = self.model(**model_inputs)
+
+        model_outputs["p_mask"] = p_mask
+        model_outputs["word_ids"] = word_ids
+        model_outputs["words"] = words
+        model_outputs["attention_mask"] = model_inputs.get("attention_mask", None)
+        return model_outputs
+
+    def postprocess(self, model_outputs, top_k=1, **kwargs):
+        if isinstance(self.model, VisionEncoderDecoderModel):
+            answers = self.postprocess_encoder_decoder(model_outputs)
+        else:
+            answers = self.postprocess_extractive_qa(model_outputs, top_k=top_k, **kwargs)
+
+        answers = sorted(answers, key=lambda x: x["score"], reverse=True)[:top_k]
+        if len(answers) == 1:
+            return answers[0]
+        return answers
+
+    def postprocess_encoder_decoder(self, model_outputs, **kwargs):
+        # postprocess
+        sequence = self.tokenizer.batch_decode(model_outputs.sequences)[0]
+        sequence = sequence.replace(self.tokenizer.eos_token, "").replace(self.tokenizer.pad_token, "")
```

Comment:
Why using `skip_special_tokens=True` in batch_decode, doesn't work ?

Reply:
I'm not sure, again I borrowed this from the example [here](https://huggingface.co/spaces/nielsr/donut-docvqa/blob/main/app.py). As I'm not an expert in Donut, I could also remove this implementation until someone more familiar with it contributes an implementation?

Reply:
Hmm yes good point @Narsil, that will also do the job. Borrowed this from the authors code, will update.

Reply:
Update: it doesn't work with skip_special_tokens=True, due to other characters (like "<" being) removed. So I'd guess we can call this method `postprocess_donut` rather than `postprocess_encoder_decoder`, as it's quite specific to Donut.

Reply:
Will rename

Reply:
Then it doesn't belong here IMO. This should be handled in the tokenizer itself.

Why are all tokens special in donut ? That doesn't seem logic. There are only very few tokens that can really be special (bos, eos, pad, unk etc..) not regular text.

Hard replacing tokens is not something that belong in the pipeline. It's super model specific so it's probably the model's own classes that should handle it.

We can accept this state in the  current PR, but we will need to refactor this at some point.

Reply:
+1

Reply:
I'll make sure to include this in the follow up tasks and will add a TODO in the code.